tests/rpc/eth/getBlockByNumber.js

```diff
@@ -0,0 +1,164 @@
+const test = require('tape')
+const sinon = require('sinon')
+const request = require('supertest')
+const { INVALID_PARAMS } = require('../../../lib/rpc/error-code')
+const { startRPC, closeRPC, createManager } = require('../helpers')
+
+function createBlockchain () {
+  const transactions = [{
+    hash: '0xc6ef2fc5426d6ad6fd9e2a26abeab0aa2411b7ab17f30a99d3cb96aed1d1055b'
+  }]
+  const block = {
+    toJSON: sinon.stub().returns({ number: 1, transactions })
+  }
+  return {
+    getBlock: sinon.stub().yields(null, block)
+  }
+}
+
+function checkError (expectedCode, expectedMessage) {
+  return function (res) {
+    if (!res.body.error) {
+      throw new Error('should return an error object')
+    }
+    if (res.body.error.code !== expectedCode) {
+      throw new Error(`should have an error code ${expectedCode}`)
+    }
+    if (expectedMessage && res.body.error.message !== expectedMessage) {
+      throw new Error(`should have an error message "${expectedMessage}"`)
+    }
+  }
+}
+
+test('call eth_getBlockByNumber with valid arguments', t => {
+  const manager = createManager(createBlockchain())
+  const server = startRPC(manager.getMethods())
+
+  const req = {
+    jsonrpc: '2.0',
+    method: 'eth_getBlockByNumber',
+    params: ['0x1', true],
+    id: 1
+  }
+
+  request(server)
```

Comment:
Nice setup with this ``supertest`` library - didn't know that yet - really sleak.